ipv-api/src/main/java/uk/gov/di/authentication/ipv/helpers/IPVCallbackHelper.java

```diff
@@ -0,0 +1,361 @@
+        auditService.submitAuditEvent(
+                IPVAuditableEvent.IPV_UNSUCCESSFUL_AUTHORISATION_RESPONSE_RECEIVED,
+                clientSessionId,
+                sessionId,
+                authenticationRequest.getClientID().getValue(),
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN);
+        var errorResponse =
+                new AuthenticationErrorResponse(
+                        authenticationRequest.getRedirectionURI(),
+                        errorObject,
+                        authenticationRequest.getState(),
+                        authenticationRequest.getResponseMode());
+        return generateApiGatewayProxyResponse(
+                302, "", Map.of(ResponseHeaders.LOCATION, errorResponse.toURI().toString()), null);
+    }
+
+    public AccountInterventionStatus getAccountInterventionStatus(
+            String internalPairwiseSubjectId) {
+        var accountInterventionStatus =
+                segmentedFunctionCall(
+                        "AIS: getAccountStatus",
+                        () ->
+                                this.accountInterventionService.getAccountStatus(
+                                        internalPairwiseSubjectId));
+        cloudwatchMetricsService.incrementCounter(
+                "AISResult",
+                Map.of(
+                        "blocked",
+                        String.valueOf(accountInterventionStatus.blocked()),
+                        "suspended",
+                        String.valueOf(accountInterventionStatus.suspended()),
+                        "resetPassword",
+                        String.valueOf(accountInterventionStatus.resetPassword()),
+                        "reproveIdentity",
+                        String.valueOf(accountInterventionStatus.reproveIdentity())));
+        return accountInterventionStatus;
+    }
+
+    public Optional<ErrorObject> validateUserIdentityResponse(UserInfo userIdentityUserInfo)
+            throws IpvCallbackException {
+        LOG.info("Validating userinfo response");
+        if (!LevelOfConfidence.MEDIUM_LEVEL
+                .getValue()
+                .equals(userIdentityUserInfo.getClaim(VOT.getValue()))) {
+            LOG.warn("IPV missing vot or vot not P2.");
+            return Optional.of(OAuth2Error.ACCESS_DENIED);
+        }
+        var trustmarkURL =
+                buildURI(configurationService.getOidcApiBaseURL().orElseThrow(), "/trustmark")
+                        .toString();
+
+        if (!trustmarkURL.equals(userIdentityUserInfo.getClaim(VTM.getValue()))) {
+            LOG.warn("VTM does not contain expected trustmark URL");
+            throw new IpvCallbackException("IPV trustmark is invalid");
+        }
+        return Optional.empty();
+    }
+
+    public APIGatewayProxyResponseEvent processReturnCode(
```

Comment:
Could you add a bit more detail to this name? I don't think it's super clear as to what it's doing, eg when it's called I wouldn't expect it to return an API gateway response (sorry I know naming is hard!)